src/receivers/mod.rs

```diff
@@ -8,32 +8,40 @@ use std::{thread, time::Duration};
 use crossbeam_channel::Sender;
 
 use crate::message::Message;
+pub use error::ReceiverError;
+pub use http::HTTP;
 pub use ping::Ping;
 
-pub struct Receiver<T>
-where
-    T: Collector,
-{
-    collector: T,
+pub struct Receiver {
+    collectors: Vec<Box<dyn Collector + Send>>,
 }
 
-impl<T> Receiver<T>
-where
-    T: Collector,
-{
-    pub fn new(collector: T) -> Receiver<T> {
-        Receiver { collector }
+impl Receiver {
+    pub fn new() -> Receiver {
+        Receiver {
+            collectors: Vec::new(),
+        }
+    }
+
+    pub fn register(&mut self, collector: impl Collector + Send + 'static) {
```

Comment:
It took me a while to understand why static lifetime annotation is needed here. Best explanation I found was this:
> Types with no references count as T: 'static. Because 'static means the reference must live as long as the entire program, a type that contains no references meets the criteria of all references living as long as the entire program (since there are no references). Think of it this way: if the borrow checker is concerned about references living long enough, then there's no real distinction between a type that has no references and a type that has references that live forever; both of them are the same for the purpose of determining whether or not a reference has a shorter lifetime than what it refers to.